kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/SubscriptionServiceImpl.java

```diff
@@ -310,10 +310,8 @@ public Subscription createSubscription(long callingNumber, Language language, Ci
             subscriberDataService.create(subscriber);
         }
 
-        // todo: switch to subscriber.getLanguage() & .getCircleName() when the ticket is fixed
-        // https://applab.atlassian.net/browse/MOTECH-1678
-        Language subscriberLanguage = (Language) subscriberDataService.getDetachedField(subscriber, "language");
-        Circle subscriberCircle = (Circle) subscriberDataService.getDetachedField(subscriber, "circle");
+        Language subscriberLanguage = subscriber.getLanguage();
```

Comment:
yea, this seems to be more of a test bug (transaction) than a real issue highlighted in 1678. it should definitely make our code much faster